### PR TITLE
new plots for tracking DQM 81x

### PR DESCRIPTION
**Pull request by mtosi**
as requested by tracking POG, 
some new plots have been added to the tracking DQM

in addition, I fix the track stopping source name by adding a dedicated list of strings

### CODE REVIEW

**Comment by mtosi** [comment]
@dmitrijus @rovere @VinInn

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
+1

For #16013 58348f5cd9a4bee14ae0f934557984d65dc689b8

Adding new tracking DQM plots. There should be no change in monitored RECO quantities.

#16014 is the 80X version of this PR, and it has already been approved by Reco.

The code changes are satisfactory, and Jenkins tests against baseline CMSSW_8_1_X_2016-09-28-1100 show the new DQM plots and no other significant differences, as expected.

File: DQM/TrackingMonitor/src/TrackAnalyzer.cc [hunk 1]
```diff
@@ -567,34 +625,33 @@ void TrackAnalyzer::bookHistosForHitProperties(DQMStore::IBooker & ibooker) {
...
-      if(StopReasonName.size() != static_cast<unsigned int>(StopReason::SIZE)) {
+      size_t StopReasonNameSize = sizeof(StopReasonName::StopReasonName)/sizeof(std::string);
+      if(StopReasonNameSize != static_cast<unsigned int>(StopReason::SIZE)) {
         throw cms::Exception("Assert") << "StopReason::SIZE is " << static_cast<unsigned int>(StopReason::SIZE)
-                                       << " but TrackAnalyzer has StopReasonName's only for "
-                                       << StopReasonName.size()
-                                       << ". Please update also TrackAnalyzer.cc near line " << __LINE__ << ".";
+                                       << " but StopReasonName's only for "
+                                       << StopReasonNameSize
+                                       << ". Please update DataFormats/TrackReco/interface/TrajectoryStopReasons.h.";
```
Comment (makortel) [inline DQM/TrackingMonitor/src/TrackAnalyzer.cc]: This check would be better implemented as a `static_assert` in DataFormats/TrackReco/interface/TrajectoryStopReasons.h

File: DataFormats/TrackReco/interface/TrajectoryStopReasons.h [hunk 2]
```diff
@@ -14,7 +14,26 @@ enum class StopReason {
...
+    "MAX_LOST_HITS",                 //  2
+    "MAX_CONSECUTIVE_LOST_HITS",     //  3
+    "LOST_HIT_FRACTION",             //  4
+    "MIN_PT",                        //  5
+    "CHARGE_SIGNIFICANCE",           //  6
+    "LOOPER",                        //  7
+    "MAX_CCC_LOST_HITS",             //  8
+    "NO_SEGMENTS_FOR_VALID_LAYERS",  //  9
+    "SEED_EXTENSION",                // 10
+    "NOT_STOPPED"                    // 11 (be careful ...)
```
Comment (makortel) [inline DataFormats/TrackReco/interface/TrajectoryStopReasons.h]: Could you improve the comment (essentially that `NOT_STOPPED` needs to be the last, and that it's index differs from the enumeration value)? I had to think a bit again what was going on :)

**Review by makortel — COMMENTED** [review]
Both of my comments can be addressed later. In addition, I think it would be good to make corresponding changes for stop reason names to TrackBuildingAnalyzer.

File: DataFormats/TrackReco/interface/TrajectoryStopReasons.h (hunk 2 above)
Comment (mtosi) [inline DataFormats/TrackReco/interface/TrajectoryStopReasons.h]: thanks @makortel,
I think I've just addressed both comments, but the one on the `static_assert`

**Comment by VinInn** [comment]
@cmsbuild , please test

**Comment by dmitrijus** [comment]
+1

**Comment by cvuosalo** [comment]
+1

For #16013 3011d8700923bd913aa968d16fe6dc04dba01b20

Second approval, after minor code changes. The new code is satisfactory, and the new Jenkins results show no problems, just the new DQM plots, as expected.

**Comment by davidlange6** [comment]
+1

**Pull request merged**